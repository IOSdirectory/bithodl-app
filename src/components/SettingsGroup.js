import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  group: {
    backgroundColor: 'white',
    borderTopColor: '#DDDDDF',
    borderBottomColor: '#DDDDDF',
    borderTopWidth: StyleSheet.hairlineWidth,
    borderBottomWidth: StyleSheet.hairlineWidth,
    marginBottom: 30
  }
});

export default class SettingsGroup extends Component {
  render() {
    return (
      <View style={[styles.group, this.props.style]}>
        {this.props.children}
      </View>
    );
  }
}

SettingsGroup.propTypes = {
  style: PropTypes.any,
  children: PropTypes.node
};
