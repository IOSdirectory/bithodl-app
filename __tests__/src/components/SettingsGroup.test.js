import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import SettingsGroup from '../../../src/components/SettingsGroup';

describe('SettingsGroup', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsGroup>
        <Text>7ec4a1e5-7bc9-4e48-99cc-304d19ebc138</Text>
      </SettingsGroup>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
