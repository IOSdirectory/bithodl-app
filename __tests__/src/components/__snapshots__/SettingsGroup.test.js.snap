// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsGroup renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderBottomColor": "#DDDDDF",
        "borderBottomWidth": 0.5,
        "borderTopColor": "#DDDDDF",
        "borderTopWidth": 0.5,
        "marginBottom": 30,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    7ec4a1e5-7bc9-4e48-99cc-304d19ebc138
  </Text>
</View>
`;
